### PR TITLE
timesync: fix wrong type for receiving timestamp in nanoseconds

### LINKED ISSUE #20482: systemd-timesyncd fails to sync time from NTP server with IPv6 address
**systemd version the issue has been seen with**
systemd v249 and main branch

**Used distribution**
 > OpenBMC

**Linux kernel version used** (`uname -a`)

 > `5.10.53`

**CPU architecture issue was seen on**
 > ARM

**Expected behaviour you didn't see**
 > systemd-timesyncd should be able to sync time from NTP server with IPv6 address

**Unexpected behaviour you saw**
 > It fails to sync time and get error message as below.

**Steps to reproduce the problem**
- Config the NTP server with IPv6 address
- Start systemd-timesyncd

**Additional program output to the terminal or log subsystem illustrating the issue**

```text
systemd-timesyncd[534]: Error receiving message, disconnecting: Exchange full
```

### PR DESCRIPTION
**Pull request by yuwata**
Hopefully fixes #20482.

### CODE REVIEW

File: src/libsystemd-network/icmp6-util.c [hunk 1]
```diff
@@ -186,7 +186,6 @@ int icmp6_receive(int fd, void *buffer, size_t size, struct in6_addr *ret_dst,
 
         /* namelen == 0 only happens when running the test-suite over a socketpair */
 
-        assert(!(msg.msg_flags & MSG_CTRUNC));
```
Comment (poettering) [inline src/libsystemd-network/icmp6-util.c:189]: why drop this?

Comment (yuwata) [reply src/libsystemd-network/icmp6-util.c:189]: `recvmsg_safe()` already handled the case.
You dropped the same check in resolved by 3691bcf3c5eebdcca5b4f1c51c745441c57a6cd1.

**Review by bluca — APPROVED** [review]
Looks correct but reporter says the issue is not fixed. Should this be held back out is it good regardless?

**Comment by bluca** [comment]
@mrc0mmand any idea about the CentOS failure? There's a java backtrace

**Comment by mrc0mmand** [comment]
> @mrc0mmand any idea about the CentOS failure? There's a java backtrace

There was a Jekins update during the night and this one slipped my radar when I was rescheduling the aborted jobs - rescheduled, thanks.

File: src/libsystemd-network/icmp6-util.c (hunk 1 above)
Comment (poettering) [reply src/libsystemd-network/icmp6-util.c:189]: indeed!

**Pull request merged**